### ISSUE TITLE
Fails to download if file already exists

### ISSUE THREAD

**Issue by pacey** [giggio/node-chromedriver#126]
I am currently getting an error when downloading chromedriver and the file already exists:

```
/tmp/deployment/application/node_modules/chromedriver/chromedriver is not writable: EACCES: permission denied, mkdir '/tmp/deployment/application/node_modules/chromedriver/chromedriver'
Downloading https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip
Saving to /tmp/chromedriver/chromedriver_linux64.zip
Received 781K...
Received 1568K...
Received 2352K...
Received 3136K...
Received 3920K...
Received 3995K total.
Extracting zip contents
Copying to target path /tmp/deployment/application/node_modules/chromedriver/lib/chromedriver
ChromeDriver installation failed { Error: EEXIST: file already exists, mkdir '/tmp/deployment/application/node_modules/chromedriver/lib/chromedriver'
at Error (native)
at Object.fs.mkdirSync (fs.js:923:18)
at copyIntoPlace (/tmp/deployment/application/node_modules/chromedriver/install.js:234:6)
at Promise._successFn (/tmp/deployment/application/node_modules/chromedriver/install.js:75:12)
at nextTickCallback (/tmp/deployment/application/node_modules/kew/kew.js:47:28)
at _combinedTickCallback (internal/process/next_tick.js:73:7)
at process._tickCallback (internal/process/next_tick.js:104:9)
errno: -17,
code: 'EEXIST',
syscall: 'mkdir',
path: '/tmp/deployment/application/node_modules/chromedriver/lib/chromedriver' }
```

**Comment by giggio** [comment]
@pacey what is the Node.js version, Npm version, OS and architecture?

**Comment by pacey** [comment]
This is running on an Amazon Elastic Beanstalk instance "64bit Amazon Linux 2017.03 v4.2.0 running Node.js". The node version is 6.11.1 and the npm version is 3.10.10

**Comment by pacey** [comment]
I think this is happening because the directory '/tmp/deployment/application/node_modules/chromedriver/lib/chromedriver' exists but for another architecture/os

**Comment by giggio** [comment]
Which is still strange, as should still copy.
It is failing on this line: https://github.com/giggio/node-chromedriver/blob/c62b7ee100198ce65ea29fff26967fb3f7aab314/install.js#L234

This is node's standard `mkdirsync` function call. It will fail only if the directory indeed exists. But in this line (2 lines before):
https://github.com/giggio/node-chromedriver/blob/c62b7ee100198ce65ea29fff26967fb3f7aab314/install.js#L232
We actually delete the directory.

So, I would guess something is keeping the directory from being excluded (ie it is busy). I just tested and `rimraf` fails silently if used in the sync version. The async version informs the error.

As `copyIntoPlace` is already a promise returning method it could change `rimraf` to `del` and use the async `del` version. At least then you will get the correct error.

For your specific problem I suggest you try to verify what is keeping the directory from being deleted. That is most likely your problem. I am closing the issue as it does not seem related with Chromedriver. But for the delete problem I opened #127 so you may get a more accurate error in the future.

**Comment by giggio** [comment]
@pacey I just fixed that connected error and published in `2.33.2`. Try again and see if you can confirm the resource busy hypothesis.

**Comment by pacey** [comment]
Thanks for looking at this, I'll give it another go tomorrow and let you know!
